tools/roxvet/analyzers/testtags/analyzer.go

```diff
@@ -0,0 +1,82 @@
+package testtags
+
+import (
+	"go/ast"
+	"path/filepath"
+	"strings"
+
+	"github.com/stackrox/rox/tools/roxvet/common"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+const doc = `Ensure that our *_test.go files include a //go:build <TAG>, otherwise they won't be run by the test target.
+You can run scripts/prepend-go-build-to-tests.sh <BUILD_TAG> with the tag corresponding to your test target, to add it
+ to all *_test.go files in the directory.'`
+
+const roxPrefix = "github.com/stackrox/rox/"
+
+// Analyzer is a analysis.Analyzer from the analysis package of the Go standard lib. [It analyzes code]
+var Analyzer = &analysis.Analyzer{
+	Name:     "testtags",
+	Doc:      doc,
+	Requires: []*analysis.Analyzer{inspect.Analyzer},
+	Run:      run,
+}
+
+// Make sure this runs only on tests within stackrox/tests, our go e2e tests
+func isTestsPackage(packageName string) bool {
+	if !strings.HasPrefix(packageName, roxPrefix) {
+		return false
+	}
+	unqualifiedPackageName := strings.TrimPrefix(packageName, roxPrefix)
+	pathElems := strings.Split(unqualifiedPackageName, string(filepath.Separator))
+	if len(pathElems) == 0 {
+		return false
+	}
+	if pathElems[0] == "tests" {
+		return true
+	}
+	return false
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
```

Comment:
> Could you simulate a failure test case for this analyzer?

Sure, [here's a minimal example](https://github.com/stackrox/stackrox/pull/12033/files#diff-38ef2c19804378eb74af209cb2d6deb1fdda4faf02f3b06c13e4055e07cdf54cR1-R11) of a test without a `//go:build` directive, [it fails the style check ](https://github.com/stackrox/stackrox/actions/runs/9987169067/job/27601019133?pr=12033#step:9:154) with
`tests/no_go_build_directive_test.go:1:1: Missing //go:build directive.` as expected.
Since the test is not including any functions from `tests/common.go` the compiler won't complain here and the analyzer is needed, this won't be caught by anything else.

Now as for having multiple `//go:build` directives, [I opened another PR](https://github.com/stackrox/stackrox/pull/12034) and this one runs into the same issue you found, and [here](https://github.com/stackrox/stackrox/actions/runs/9987357040/job/27601572517?pr=12034#step:6:7) I found something interesting in the output, it says 
`multiple_go_build_directives_test.go: multiple //go:build comments` which is not a comment from my analyzer, so some other linter step already catches this. In which case I think it's fine for me to remove the check for multiple `//go:build` directives from my analyzer and we are good to go.

Reply:
> do we need to have these checks?

So in conclusion:
Yes for making sure **a** `//go:build` directive is present in `*_test.go` files within `stackrox/tests`
No for making sure there aren't multiple `//go:build` directives